FSW/headers/memUtil.hpp

```diff
@@ -0,0 +1,17 @@
+#ifndef MEMUTIL_H
```

Comment:
Are these header files just for convivence of not included all the declarations/definitions/libraries in the actual .cpp files?

Reply:
In C/C++, there are two things required to successfully compile a function: the function declaration and the function definition. When using multi-file compilation (as we are), headers are a way to ensure that all of the function declarations are preprocessed before the compiler gets to any function calls inside function definitions. If the compiler arrives at a function call before it has processed the definition, then it will throw an error and fail to compile. Using `#include <header.hpp>` ensures that all function definitions are seen by the compiler before the function calls.